src/parser/Class_Parser.hh

```diff
@@ -0,0 +1,411 @@
+ * an error in the input syntax, as when a keyword is encountered when
+ * a numerical value is expected.  In this case, an exception is thrown
+ * that is caught by the Class_Parser_Base, which then attempts error
+ * recovery as described above.  The report_semantic_error indicates
+ * syntactically correct input that is nonetheless unacceptable, as when a
+ * numerical value appears where it is expected but violates some constraint,
+ * such as positivity.  The error is reported but input processing is not
+ * interrupted. In effect, this is an error with immediate recovery.
+ *
+ * No general implementation is provided. The developer wishing to make his
+ * class parsable must specialize this template for his class. This should
+ * almost always be derived from the Class_Parser_Base template, which
+ * provides the necessary boilerplate. The final specialization must define
+ * the following members:
+ *
+ *   Class_Parser()
+ *
+ * Default constructor. The constructor should set all data members to sentinel
+ * values indicating that the corresponding parameters have not been parsed yet.
+ * Where possible, the sentinel value should be one that doesn't make
+ * sense for the parameter in question, such as 0 for a dimension that
+ * must be nonzero, or a negative value for a parameter that must be
+ * positive.  This ensures that the sentinel value can be distinguished
+ * from any value a user might specify in an input "deck." If the parameter can
+ * plausibly have any value, then a bool sentinel flag should accompany the
+ * parameter and be initialized in the constructor to false (for not yet
+ * parsed).
+ *
+ * The constructor should also carry out any other preparations that are
+ * necessary before parsing.  The constructor may optionally have a context
+ * argument to provide any contextual information required to know how to parse
+ * the "deck".
+ *
+ *   void check_completeness(Token_Stream &);
+ *
+ * This function is called to check that all required specifications were
+ * found in the parsed token stream.  This function must call \c
+ * Token_Stream::report_semantic_error at least once if a required
+ * specification is missing.
+ *
+ * If default values are permitted for some parameters (which is not
+ * recommended), then they should be applied here.
+ *
+ *   Class *create_object();
+ *
+ * Create the object from the parsed fields.  This function should have no
+ * preconditions that are not guaranteed by a preceding successful call to
+ * check_completeness. create_object will be called only if no errors
+ * were detected in the input token stream.
+ *
+ * In addition to these mandatory members, any nontrivial Class_Parser
+ * will also define data members to hold the parsed specifications, and parse
+ * functions of the form
+ *
+ * void parse_parameter(Token_Stream &, int);
+ *
+ * A nontrivial Class_Parser should define a member
+ *
+ * static Class_Parser_Keyword<Class> const raw_table[];
```

Comment:
You will need to escape the angle brackets in the documentation block.  Doxygen thinks you are inserting html commands:

```
Class_Parser_Keyword \<Class\>
```

Please fix here and below.

Reply:
I'll try wrapping this section in \code and see if it makes it better.